Content.Client/Chat/ChatBox.cs

```diff
@@ -34,96 +72,432 @@ public class ChatBox : Control
+        {
+            foreach (var child in _filterVBox.Children)
             {
-                Text = Loc.GetString("All"),
-                Name = "ALL",
-                HorizontalExpand = true,
-                HorizontalAlignment = HAlignment.Right,
-                ToggleMode = true,
-            };
+                if (child is not ChannelFilterCheckbox checkbox) continue;
+                checkbox.OnToggled -= OnFilterCheckboxToggled;
+            }
+        }
 
-            LocalButton = new Button
+        private void UnsubChannelItems()
+        {
+            foreach (var child in _channelSelectorHBox.Children)
             {
-                Text = Loc.GetString("Local"),
-                Name = "Local",
-                ToggleMode = true,
-            };
+                if (child is not ChannelItemButton button) continue;
+                button.OnPressed -= OnChannelSelectorItemPressed;
+            }
+        }
+
 
-            OOCButton = new Button
+        /// <summary>
+        /// Update the available filters / selectable channels and the current filter settings using the provided
+        /// data.
+        /// </summary>
+        /// <param name="selectableChannels">channels currently selectable to send on</param>
+        /// <param name="filterableChannels">channels currently able ot filter on</param>
+        /// <param name="channelFilters">current settings for the channel filters, this SHOULD always have an entry if
+        /// there is a corresponding entry in filterableChannels, but it may also have additional
+        /// entries (which should not be presented to the user)</param>
+        /// <param name="unreadMessages">unread message counts for each disabled channel, values 10 or higher will show as 9+</param>
+        public void SetChannelPermissions(List<ChatChannel> selectableChannels, IReadOnlySet<ChatChannel> filterableChannels,
+            IReadOnlyDictionary<ChatChannel, bool> channelFilters, IReadOnlyDictionary<ChatChannel, byte> unreadMessages)
+        {
+            _selectableChannels = selectableChannels;
+            // update the channel selector
+            UnsubChannelItems();
+            _channelSelectorHBox.RemoveAllChildren();
+            foreach (var selectableChannel in ChannelSelectorOrder)
             {
-                Text = Loc.GetString("OOC"),
-                Name = "OOC",
-                ToggleMode = true,
-            };
+                if (!selectableChannels.Contains(selectableChannel)) continue;
+                var newButton = new ChannelItemButton(selectableChannel);
+                newButton.OnPressed += OnChannelSelectorItemPressed;
+                _channelSelectorHBox.AddChild(newButton);
+            }
+            // console channel is always selectable and represented via Unspecified
+            var consoleButton = new ChannelItemButton(ChatChannel.Unspecified);
+            consoleButton.OnPressed += OnChannelSelectorItemPressed;
```

Comment:
Not sure if this needs to be a field so that it can be unsubscrievd in `UnsubChannelItems()`

Reply:
Ignoring this per PJB's comment on discord.